compiler/test/runnable/test23545.d

```diff
@@ -0,0 +1,16 @@
+export int fooexp() { return 1; } // definition  => dllexport
+export int fooimp();   // declaration => dllimport
+export __gshared int aaa = 2;    // definition  => dllexport
```

Comment:
Won't this still be problematic when using it with D header generation `-H`?

```
$ cat test.d
export __gshared int aaa = 2;

$ dmd -H -c test.d
$ cat test.di

// D import file generated from 'test.d'
export __gshared int aaa = 2;
```

Reply:
Right now hdrgen isn't suitable for shared libraries.

So we can ignore this, for the time being, it's got a lot of work ahead of it.